packages/middleware-endpoint/src/adaptors/createConfigValueProvider.ts

```diff
@@ -24,6 +24,13 @@ export const createConfigValueProvider = <Config extends Record<string, unknown>
     }
     return configValue;
   };
+  if (configKey === "credentialScope" || canonicalEndpointParamKey === "CredentialScope") {
+    return async () => {
+      const credentials = typeof config.credentials === "function" ? await config.credentials() : config.credentials;
```

Comment:
For `credentials`, I think we may need to add an undefined check since `credentials` could possibly be undefined.

Though, I am not sure where the `createConfigValueProvider()` function is used in the request lifecycle.

Reply:
That was a miss from my side, LGTM.